File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/SeriesV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;

/**
 * Canonical series/franchise descriptor.
 */
public record SeriesV1(
        String seriesId,
        String name,
        String normalizedName,
        String description,
        String mediaTypeHint,
        long createdAtEpochMs,
        long updatedAtEpochMs
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/SeriesV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract

data class SeriesV1(
    val seriesId: String,
    val name: String,
    val normalizedName: String,
    val description: String,
    val mediaTypeHint: String,
    val createdAtEpochMs: Long,
    val updatedAtEpochMs: Long
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
